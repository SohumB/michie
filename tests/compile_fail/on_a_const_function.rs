use michie::memoized;

#[memoized(key_expr = ())]
const fn f() -> () {
    ()
}

fn main() {}
