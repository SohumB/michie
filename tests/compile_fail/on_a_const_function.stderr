error: const functions not allowed
 --> tests/compile_fail/on_a_const_function.rs:4:1
  |
3 | #[memoized(key_expr = ())]
  | -------------------------- in this procedural macro expansion
4 | const fn f() -> () {
  | ^^^^^
  |
  = note: this error originates in the attribute macro `memoized` (in Nightly builds, run with -Z macro-backtrace for more info)
